### ISSUE TITLE
Dataset problem: models with non-trivial problems with Ofx or Omon.

### ISSUE THREAD

**Issue by ledm** [ESMValGroup/ESMValCore#1314]
I generated a recipe that looks at every `tos` (Temperature at the ocean surface) dataset in the historical and Tier1 SSP datasets. Pretty much every model in CMIP6 has provided this dataset. 

Elsewhere in the  the recipe, I take a spatial average using `area_statistics` preprocessor, `operator: mean`, which of course, requires the `areacello` Ofx dataset. 

The preprocessor is: 
```
    area_statistics:
      fx_variables:
      - {activity: CMIP,  grid: gn, mip: Ofx, short_name: areacello, exp: '*', ensemble: '*'}
      operator: mean
```
(which requires https://github.com/ESMValGroup/ESMValCore/pull/1082).

I've been working through these and a handful of models just need some minor fixes. These models needs the `fix_ocean_depth_coord` on all Omon data.

- cnrm_cm6_1
- noresm2_lm.py
- noresm2_mm.py

However, some models simply need to be excluded at the moment. The following models can not be used for the reason supplied: 

```
skip_models  = [      # because
    'AWI-CM-1-1-MR',  # triangular grid
    'AWI-ESM-1-1-LR', # triangular grid
    'BCC-CSM2-MR',    # No fx files.
    'CAMS-CSM1-0',    # No fx files.
    'CMCC-CM2-HR4',   # No fx files.
    'GISS-E2-1-G-CC', # No fx files.
    'GISS-E2-1-H',    # No fx files.
    'GFDL-CM4',       # High resolution  (1000x1000 grid). - also problem with lat/lon in areacello.
    'GFDL-ESM4',      #  problem with lat/lon in areacello.
    'IITM-ESM',       # No longitude bounds.
    'FGOALS',         # No longitude bounds in irregular grid.
    'MIROC-ES2L',     # SSP data contains loads of other variables.
    'NorESM2-LM',     # Depth provided in weird units (density)
    'NorESM2-MM',     # Depth provided in weird units (density)
    'r13i1p1f2',      # UKESM r13 also doesn't work as it's got the wrong grid.
    ]
```
I already wrote this as a discussion here, but @valeriupredoi suggested creating an issue as well: https://github.com/ESMValGroup/ESMValTool/discussions/2308

I have a feeling that some of this could be fixed, but others are fairly intractable at the moment. Happy to be proven wrong.

**Comment by zklaus** [comment]
Since this seems to touch many different topics and datasets, which likely will translate to a number of different issues, I suggest focusing on the discussion at ESMValGroup/ESMValTool#2308 for now.

**Comment by valeriupredoi** [comment]
yeah this issue is good to have once we've reduced the above list to the models that have wonky data/metadata ie when we've implemented all the tricks that take care of fx-related issues in our code :+1:
